crates/holochain_persistence_lmdb/src/txn/mod.rs

```diff
@@ -0,0 +1,30 @@
+use holochain_persistence_api::cas::*;
+use holochain_persistence_api::eav::*;
+use holochain_persistence_api::txn::*;
+use rkv::Writer;
+use crate::common::LmdbInstance;
+
+
+struct LmdbCursor<'prim_env, 'staging_env> {
+    // rw cursor for the staging db
+    // a reference to the staging db itself?
+    // a ro cursor over primary db
+    // reference to primary db
+    // lazy write cursor to primary db
+    primary_db_writer: FnOnce<Writer<'prim_env>>,
+    staging_db_writer: FnOnce<Writer<'staging_env>>
+    primary_db_reader: FnOnce<Reader>,
+    staging_db_reader: FnOnce<Reader>,
+    staging_db: FnOnce<LmdbInstance>,
+    primary_db: FnOnce<LmdbInstance>,
```

Comment:
This raises a question for me: does LMDB support upgrading a read cursor to a write cursor? If not, and we have to open a separate write cursor, then I guess we have to keep around the read cursor as well, because we are starting the write transaction at a time when the snapshot will be different from the read transaction. So we are using the read cursor strictly for reads and the write cursor strictly for writes, even though the write cursor can read too (is that true?)

1. start read cursor R
2. read some stuff via R
3. some *other* write cursor X modifies DB
4. R cannot read that new write since it happened before the cursor was opened
5. Now we want to write, so we open a new write cursor, W
6. W has a more recent snapshot than R, so we can't use it for reads

Reply:
This code is outdated- but the current version only opens read handles on the primary database as is needed. The commit at the end establishes the write handle.